src/scripts/reducers/VideosReducer.js

```diff
@@ -19,37 +20,9 @@ import {
   VIDEOS_FETCH_SUCCESS
 } from 'constants/ActionConstants'
 import VideoRecord from 'records/VideoRecords'
-import {
-  ResultStatusRecord,
-  AsyncTaskStatusRecord,
-  DataStatusRecord
-} from 'records/AsyncTaskStatusRecord'
+import { AsyncTaskStatusRecord } from 'records/AsyncTaskStatusRecord'
 import type { Action, VideoRecordMap } from 'types/ApplicationTypes'
 
-const castRecordFromDbAsImmutable = function (videoProps) {
-  return new VideoRecord({
-    ...videoProps,
-    thumbnails:
-      (videoProps.transcodingStatus &&
-        videoProps.transcodingStatus.data.result &&
-        videoProps.transcodingStatus.data.result.screenshots) ||
-      [],
-    uploadStatus: new AsyncTaskStatusRecord({
-      ...videoProps.uploadStatus,
-      data: new DataStatusRecord(videoProps.uploadStatus.data)
-    }),
-    transcodingStatus: new AsyncTaskStatusRecord({
-      ...videoProps.transcodingStatus,
-      data: new DataStatusRecord(videoProps.transcodingStatus.data)
-    }),
-    // we reset the storage status from the db, because it will be outdated in any case
-    storageStatus: new AsyncTaskStatusRecord()
-    // storageStatus: new AsyncTaskStatusRecord({
-    //   ...videoProps.storageStatus,
-    //   data: new DataStatusRecord(videoProps.storageStatus.data)
-    // })
-  })
-}
```

Comment:
This is no longer needed now that the record constructors handle this themselves